src/wallet.ts

```diff
@@ -461,18 +461,82 @@ export class Wallet {
       changeMethods: ['make_offer'],
     })
 
+    const setPrice = price || list.price
+
     // @ts-ignore: method does not exist on Contract type
     await contract.make_offer(
       {
         token_key: list.token.id,
-        price: price || list.price,
+        price: setPrice,
         timeout: { Hours: 72 },
       },
       GAS,
-      list.price
+      setPrice
     )
   }
 
+  /**
+   * Make an offer to a token.
+   * @param tokenId
+   * @param price
+   */
+  public async makeOffer(tokenId: string, price: string): Promise<void> {
+    const account = this.activeWallet?.account()
+    const accountId = this.activeWallet?.account().accountId
+    const GAS = new BN('300000000000000')
+
+    if (!account || !accountId) throw new Error('Account is undefined.')
+    if (!tokenId) throw new Error('Please provide a tokenId')
+
+    const contract = new Contract(account, MARKET_ACCOUNT, {
+      viewMethods: [],
+      changeMethods: ['make_offer'],
+    })
+
+    // @ts-ignore: method does not exist on Contract type
+    await contract.make_offer(
+      {
+        token_key: tokenId,
+        price: price,
+        timeout: { Hours: 72 },
+      },
+      GAS,
+      price
+    )
+  }
+
+  /**
+   * Make an offer to a token.
+   * @param tokenId
+   * @param price
+   */
+  public async acceptAndTransfer(
```

Comment:
this function is the final pull from the lister and transfers the NFT, works like a charm, thanks @thor314